ICSharpCode.Decompiler/TypeSystem/IParameter.cs

```diff
@@ -105,4 +105,34 @@ public interface IParameter : IVariable
 		/// </summary>
 		IParameterizedMember? Owner { get; }
 	}
+
+	internal static class IParameterExtensions
+	{
+		public static bool GetDefaultValueAssignmentAllowed(this IParameter parameter)
+		{
+			if (!DefaultValueAssignmentAllowedIndividual(parameter))
+				return false;
+
+			if (parameter.Owner == null)
+				return true;
```

Comment:
if the parameter belongs to an anonymous method, we'd still have to figure out whether we can use assignment syntax, I think...

Also I am not sure, whether the `Owner` is actually null in practice, I'd expect the `IParameter` instance that is used for the lambda parameters just to be "owned" by the compiler-generated method from which the lambda was constructed. But I might be wrong.

Reply:
I can write a unit test?

Reply:
Yes, you could do that... writing a pretty test that contains a lambda with parameter attributes should be possible in recent versions of C#

Looking at the code that actually creates the anonymous method expressions/lambdas, it seems I remember it correctly: https://github.com/icsharpcode/ILSpy/blob/df1f1051d4fd69595b00fbe829b52826505e4ea6/ICSharpCode.Decompiler/CSharp/ExpressionBuilder.cs?plain=1#L2406C1-L2406C75

So maybe the documentation needs to be updated :)

Reply:
> Also I am not sure, whether the `Owner` is actually null in practice, I'd expect the `IParameter` instance that is used for the lambda parameters just to be "owned" by the compiler-generated method from which the lambda was constructed. But I might be wrong.

This intuition was correct.

Reply:
> So maybe the documentation needs to be updated :)

No, I think, we should keep it that way, otherwise it might break all code that uses `DefaultParameter`s...